client/grpc_query.go

```diff
@@ -120,3 +119,22 @@ func (ctx Context) Invoke(grpcCtx gocontext.Context, method string, req, reply i
 func (Context) NewStream(gocontext.Context, *grpc.StreamDesc, string, ...grpc.CallOption) (grpc.ClientStream, error) {
 	return nil, fmt.Errorf("streaming rpc not supported")
 }
+
+// selectHeight returns the height chosen from client context and grpc context.
+// If exists, height extracted from grpcCtx takes precedence.
+func selectHeight(clientContext Context, grpcCtx gocontext.Context) (int64, error) {
```

Comment:
Are we choosing 0 when neither is set? If so maybe lets set this more explicitly?

Should we be having input validation for heights > latest block height?

Reply:
Zero is guaranteed to be returned in that case by `var height int64` which is being initialized to 0. There is a test to cover this case as well. How are you suggesting making this more explicit? With `height := int64(0)` or adding a relevant comment to the spec? Just making sure I understand what you mean

The validation for `heights > latest block height` is done at the server layer. It would be challenging to have that information here in client.